src/lib.rs

```diff
@@ -23,38 +24,46 @@ const FILL_PER_ENTERTAINMENT :u64 = 1000;  // - how much Tamagotchi becomes happ
 static mut TAMAGOTCHI: Option<Tamagotchi> = None;
 
 impl Tamagotchi {
-    
+    fn get_current_feed(&mut self) -> u64  {
+        let _current_block = exec::block_height() as u64;
+        let  hungry_val = (_current_block - self.fed_block)*HUNGER_PER_BLOCK;
+        self.fed  =  if self.fed > hungry_val {self.fed -hungry_val } else { 0 };
```

Comment:
You can use `min` / `max` here:
```suggestion
        self.fed -= hungry_val.min(self.fed);
```